File: matlab/P2L2Plot.m
function fh=P2L2Plot(parms,fdv)

Pyr=1;
Lac=2;

%Unpack variables:
for ii=1:length(parms)
    x.(fdv.fitvars{ii})=parms(ii);
end
for ii=1:length(fdv.knowns)
    x.(fdv.knowns{ii})=fdv.knownvals(ii);
end

NSeg=fdv.NSeg;

[EV,IV,vb,Mzev,Mziv] = P2L2(parms,fdv);
tot=(IV*vb)+EV*(1-vb);
Mz =(Mziv*vb)+Mzev*(1-vb);
resid=P2L2Err(parms,fdv);
resn=sqrt(sum(resid.^2));

fh=gcf;
% First plot Mz
%yyaxis left 
subplot(2,1,1)
plot(fdv.taxis(NSeg:NSeg:end),Mz(Pyr,:),'g-',...
    fdv.taxis(NSeg:NSeg:end),Mz(Lac,:),'b-',...
    'linewidth',2)
legend('Pyr','Lac','Location','Best')
ylabel('Mz')
title(sprintf('%s: L2Norm = %10.4e',fdv.Name,resn))
%yyaxis right
subplot(2,1,2)
plot(fdv.taxis(NSeg:NSeg:end),tot(Pyr,:),'g-',...
    fdv.taxis(NSeg:NSeg:end),tot(Lac,:),'b-',...
    fdv.taxis(NSeg:NSeg:end),fdv.data(Pyr,:),'gx',...
    fdv.taxis(NSeg:NSeg:end),fdv.data(Lac,:),'bx',...
    'linewidth',2,'MarkerSize',8)
title(sprintf('k_p_l=%5.3f, k_v_e=%5.3f, v_b=%6.3e',x.kpl,x.kve,x.vb))
ylabel('Mxy')
xlabel('Time (s)')
legend('Pyr','Lac','Location','Best')

File: python/P2L2Plot.py
import matplotlib.pyplot as plt
import numpy as np
import P2L2

def plot(resid, parms, fdv):
    '''
    A function to plot the result of TestP2L1.py

    The customizability of this function is fairly limited without editing.

    L2Norm is defined as the norm of the vector residuals array.
    '''
    pyr = 0 
    lac = 1

    EV , IV, vb, Mzev, Mziv = P2L2.P2L2(parms, fdv)
    tot = IV*vb + (1-vb)*EV
    Mz = (Mziv*vb) + Mzev*(1-vb)

    residNorm = np.linalg.norm(resid)

    fig, ax = plt.subplots(2, figsize = (12, 6))
    fig.suptitle('Vertically stacked plot')

    ax[0].title.set_text(fdv['Name'] + ': L2 Norm = ' + str(residNorm))
    ax[0].plot(fdv['taxis'], Mz[pyr], color = 'green', label = 'Pyr')
    ax[0].plot(fdv['taxis'], Mz[lac], color = 'blue', label = 'Lac')

    ax[1].title.set_text('kpl = ' + str(parms[0])
        + ', kve = ' + str( fdv['knowns']['kve'] )
        + ', vb = ' + str( vb ))
    ax[1].plot(fdv['taxis'], tot[pyr], color = 'green', label = 'Pyr')
    ax[1].plot(fdv['taxis'], tot[lac], color = 'blue', label = 'Lac')
    ax[1].scatter(fdv['taxis'], fdv['data'][pyr], color = 'green', marker = 'x')
    ax[1].scatter(fdv['taxis'], fdv['data'][lac], color = 'blue', marker = 'x')

    ax[0].legend(fontsize = 'x-large')
    ax[0].grid()
    ax[0].set_ylabel('Mz')
    
    ax[1].set_xlabel('Time (s)')
    ax[1].grid()
    ax[1].set_ylabel('Mxy')

    plt.subplots_adjust(hspace = .5)
    plt.show()

    return fig